core/rio/api/src/main/java/org/eclipse/rdf4j/rio/helpers/BasicWriterSettings.java

```diff
@@ -113,6 +113,17 @@
 			"org.eclipse.rdf4j.rio.encode_rdf_star",
 			"Encodes RDF* triples to special IRIs for compatibility with RDF", Boolean.TRUE);
 
+	/**
+	 * Boolean setting for writer to determine whether it should remove the rdf:nodeID attribute from blank nodes.
+	 * <p>
+	 * Defaults to False.
+	 * <p>
+	 * Can be overridden by setting system property {@code org.eclipse.rdf4j.rio.compact_xml}.
+	 */
+	public static final RioSetting<Boolean> COMPACT_XML = new BooleanRioSetting(
+			"org.eclipse.rdf4j.rio.compact_xml",
+			"To generate a compact XML i.e. without Blank Node ID.", Boolean.FALSE);
+
```

Comment:
I like this, but perhaps a better place for this is `XMLWriterSettings`, since this configuration option is only relevant for XML.

Reply:
sure.